third_party/sigs.k8s.io/controller-runtime/pkg/dynamiccache/internal/informers_map.go

```diff
@@ -138,33 +139,33 @@ func (e *MapEntry) Start(stop <-chan struct{}) {
 	e.Informer.Run(internalStop)
 }
 
-// Start calls Run on each of the informers and sets started to true.  Blocks on the stop channel.
+// Start calls Run on each of the informers and sets started to true.  Blocks on the context.
 // It doesn't return start because it can't return an error, and it's not a runnable directly.
-func (ip *specificInformersMap) Start(stop <-chan struct{}) {
+func (ip *specificInformersMap) Start(ctx context.Context) {
 	func() {
 		ip.mu.Lock()
 		defer ip.mu.Unlock()
 
 		// Set the stop channel so it can be passed to informers that are added later
-		ip.stop = stop
+		ip.stop = ctx.Done()
 
 		// Start each informer
-		for _, entry := range ip.informersByGVK {
-			go entry.Start(stop)
+		for _, informer := range ip.informersByGVK {
+			go informer.Informer.Run(ctx.Done())
```

Comment:
This patch got reverted - we should start the MapEntry, not the informer directly. That will allow stopping each particular informer separately.

Reply:
thanks! reverted back to using MapEntry